### PR TITLE
Default threads based on available processors

### PR DESCRIPTION
**Pull request by psiinon**
Part of #7663 - the rest are in zap-extensions

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

**Comment by psiinon** [comment]
Added unit tests (should have done that in the first PR;)

**Comment by psiinon** [comment]
Updated to not limit the number of threads in the GUI

**Review by kingthorin — APPROVED** [review]

**Comment by psiinon** [comment]
Oops - just found out this causes the desktop to hang 😱 
It turns out PositiveValuesSlider doesnt like max int ... I'll change it to a number spinner...

**Comment by psiinon** [comment]
OK, think we're good now 😉

File: zap/src/main/java/org/zaproxy/zap/extension/ascan/OptionsScannerPanel.java [hunk 1]
```diff
@@ -441,19 +427,12 @@ private JSlider getSliderHostPerScan() {
...
-                    new ChangeListener() {
-                        @Override
-                        public void stateChanged(ChangeEvent e) {
-                            setLabelThreadsPerHostValue(getSliderThreadsPerHost().getValue());
-                        }
-                    });
+    private ZapNumberSpinner getSpinnerThreadsPerHost() {
+        if (spinnerThreadsPerHost == null) {
+            spinnerThreadsPerHost =
+                    new ZapNumberSpinner(0, Constant.getDefaultThreadCount(), Integer.MAX_VALUE);
```
Comment (thc202) [inline zap/src/main/java/org/zaproxy/zap/extension/ascan/OptionsScannerPanel.java:433]: Minimum could be 1.

**Comment by psiinon** [comment]
Hopefully fixed all of those 😁

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**